### PR TITLE
Authorize only staff users to modify database (create/edit/destroy)

### PR DESCRIPTION
**Pull request by kyriakipapadimitriou**
Iter 4.2
Users have different types (staff, student, and coach). Only staff should be able to create, edit, and destroy users, orgs, apps, engagements, and iterations.
I also added a brief description to the 'Create' view summarizing the purpose of the page.
Each controller has a skip before filter checking the type of user. If a user is not authorized to modify the database, they cannot create, edit, and destroy users, orgs, apps, engagements and iterations. 
Going forward cucumber tests should include type of user in order to be able to pass the check in the controller actions.
UPDATE: The application's controller had a long method with if statements which was failing Travis' complexity limit. I therefore broke it down to smaller parts.

### CODE REVIEW

**Review by kimhoang73 — APPROVED** [review]
the filters look very nice and dry! and i like how you decided to give more customized messages to the user, telling them only staff can do certain actions and not just redirecting them without any guidance. this definitely improves the UX!

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -7,11 +7,38 @@ class ApplicationController < ActionController::Base
...
+    elsif @@name_path.first(5) == "/orgs"
+      redirect_helper(orgs_path, 'Error: Only Staff can create, edit and destroy orgs')
+    elsif @@name_path.first(5) == "/user"
+      redirect_helper(users_path, 'Error: Only Staff can create and edit users')
+    elsif @@name_path.first(5) == "/crea"
+      redirect_helper(apps_path, 'Error: Only Staff can create apps, orgs, users')
+    elsif @@name_path.first(5) == "/enga"
+      parts = @@name_path.split("/")
+      path = "/" + parts[1] + "/" + parts[2] + "/" + parts[3]
+      redirect_helper(path, 'Error: Only Staff can create, edit and destroy apps, engagements and iterations')
```
Comment (Jldeatrick) [inline app/controllers/application_controller.rb:35]: Should probably get DRY'd out during a future iteration, but this is definitely a complicated method to implement so good job getting it done

**Review by Jldeatrick — APPROVED** [review]
Well done! We've been wanting to implement this since the first iteration so great job getting it done!

**Review by csungwon — APPROVED** [review]
looks great! The implementation is dry and accurate! Good that you put all overlapping behavior into application_controller.

**Pull request merged**